main.tf

```diff
@@ -182,6 +182,58 @@ resource "aws_iam_role_policy_attachment" "firehose" {
   policy_arn = aws_iam_policy.firehose[0].arn
 }
 
+data "aws_iam_policy_document" "firehose_s3_delivery_kms" {
+  count = var.create_route53_subzone && var.create_route53_query_log && local.iam_role_create_firehose && var.route53_query_log_bucket_kms_key != null ? 1 : 0
+
+  version = "2012-10-17"
+
+  statement {
+    sid    = "KmsActions"
+    effect = "Allow"
+
+    actions = [
+      "kms:Decrypt",
```

Comment:
Don't you need encrypt action as well?

Reply:
apparently not! didn't even need decrypt!